### PR TITLE
Bring in brotli support from PR on BrowserMob

### PR DESCRIPTION
**Pull request by ericbeland**
This is a port of the PR for brotli support on the BrowserMob proxy.

### CODE REVIEW

File: browserup-proxy-core/build.gradle [hunk 1]
```diff
@@ -59,6 +59,8 @@ uploadArchives {
 dependencies {
     implementation project(':browserup-proxy-mitm')
 
+    compile group: 'org.brotli', name: 'dec', version: '0.1.2'
```
Comment (dpowell) [inline browserup-proxy-core/build.gradle]: Change to `implementation` and alphabetize in below amongst the others of that type. Probably also use the shorthand convention.

`implementation 'org.brotli:dec:0.1.2'`

File: browserup-proxy-core/src/main/java/com/browserup/bup/filters/ServerResponseCaptureFilter.java [hunk 2]
```diff
@@ -29,7 +29,7 @@
  */
 public class ServerResponseCaptureFilter extends HttpFiltersAdapter {
     private static final Logger log = LoggerFactory.getLogger(ServerResponseCaptureFilter.class);
-
+    private static final String BROTLI_COMPRESSION = "br";
```
Comment (dpowell) [inline browserup-proxy-core/src/main/java/com/browserup/bup/filters/ServerResponseCaptureFilter.java]: Preserve the blank line separating these from the below comment.

File: browserup-proxy-core/src/main/java/com/browserup/bup/util/BrowserUpHttpUtil.java [hunk 3]
```diff
@@ -113,6 +115,42 @@ public static long getHeaderSize(HttpHeaders headers) {
     }
 
     /**
+     * Decompresses the brotli byze stream
+     *
+     * @param fullMessage brotli byte stream to decompress
+     * @return decompressed bytes
+     * @throws DecompressionException thrown if the fullMessage cannot be read or decompressed for any reason
+     */
+    public static byte[] decompressBrotliContents(byte[] fullMessage) throws DecompressionException {
```
Comment (dpowell) [inline browserup-proxy-core/src/main/java/com/browserup/bup/util/BrowserUpHttpUtil.java]: Probably should rename the above `decompressContents` to `decompressGZIPContents`.

Comment (ericbeland) [reply browserup-proxy-core/src/main/java/com/browserup/bup/util/BrowserUpHttpUtil.java]: Fixed in latest.

File: browserup-proxy-core/src/main/java/com/browserup/bup/filters/ServerResponseCaptureFilter.java (hunk 2 above)
Comment (ericbeland) [reply browserup-proxy-core/src/main/java/com/browserup/bup/filters/ServerResponseCaptureFilter.java]: fixed in latest

File: browserup-proxy-core/build.gradle (hunk 1 above)
Comment (ericbeland) [reply browserup-proxy-core/build.gradle]: fixed

**Pull request merged**